rsts/deployment/cluster_config/performance.rst

```diff
@@ -13,16 +13,49 @@ The video below contains an overview of the Flyte architecture, what is meant by
 Scaling up FlytePropeller
 ==========================
 `FlytePropeller <https://pkg.go.dev/github.com/flyteorg/flytepropeller>`_ is the core engine of Flyte that executes the workflows for Flyte. It is implemented as a `controller <https://kubernetes.io/docs/concepts/architecture/controller/>`__ in Kubernetes.
+It is designed as a `Kubernetes Controller <https://kubernetes.io/docs/concepts/architecture/controller/>`_, where the desired state is specified as a FlyteWorkflow `Custom Resource <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>`_.
+
+One of the base assumptions of FlytePropeller is that every workflow is independent and can be executed by a completely distinct process, without a need for communication with other processes. On the flip-side, one workflow is tracks the dependencies using a DAG structure and hence constantly needs synchronization.
+Currently each workflow is executed by exactly one process and one thread at a given time. Each workflow is constantly evaluated in an event-loop and uses the specification to cause an effect on the K8s cluster or remote services. The actual effect is actually caused by FlytePropeller plugins.
```

Comment:
This is difficult to understand. Maybe something like:

Currently, FlytePropeller executes Workflows by using an event loop to periodically track and amend the execution status. Within each iteration, a single thread requests the state of Workflow nodes and performs operations (i.e., scheduling node executions, handling failures, etc) to gracefully transition a Workflow from the observed state to the desired state (i.e., Success). Consequently, actual node executions are performed by various FlytePlugins, a diverse collection of operations spanning k8s and other remote services, and FlytePropeller is only responsible for effectively monitoring and managing these executions.

Reply:
i am using your sentence :)